x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/action/GetCheckpointAction.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.transform.action;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.IndicesRequest;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+
+public class GetCheckpointAction extends ActionType<GetCheckpointAction.Response> {
+
+    public static final GetCheckpointAction INSTANCE = new GetCheckpointAction();
+
+    // note: this is an admin action, it must be called with user headers
+    public static final String NAME = "cluster:admin/transform/checkpoint";
+
+    private GetCheckpointAction() {
+        super(NAME, GetCheckpointAction.Response::new);
+    }
+
+    public static class Request extends ActionRequest implements IndicesRequest {
```

Comment:
This request also needs to implement `IndicesRequest.Replaceable` so the indices are expandable for Authorization.